lumicks/pylake/channel.py

```diff
@@ -583,8 +591,17 @@ def stop(self):
             raise IndexError("End of empty time series is undefined")
 
     @property
-    def _timesteps(self):
-        return np.unique(np.diff(self.timestamps)).astype(int)
+    def sample_rate(self) -> Union[float, None]:
+        """The data frequency or `None` if it is variable"""
+        timestep = self._timesteps
+        if len(timestep) > 1:
+            return None
+        else:
+            return 1e9 / timestep[0]  # Hz
```

Comment:
You could omit the `None` case, since that's the default return. No need to explicitly return it.
```suggestion
        if len(timestep) == 1:
            return 1e9 / timestep[0]  # Hz
```